packages/headless/src/database/mongodb/cart/index.ts

```diff
@@ -0,0 +1,115 @@
+import { Cart, Item } from '../../../entity/cart';
+import { ICartDatabase } from '../../../modules/cart/repositories/cart.database.interface';
+import { CartModel } from './cart.model';
+import { Injectable } from '@nestjs/common';
+
+@Injectable()
+export class CartDatabase implements ICartDatabase {
+  //   constructor() {}
+
+  async addItemByUserId(userId: string, item: Item): Promise<Cart | null> {
+    const cart = await CartModel.findOneAndUpdate(
+      { user: userId },
+      { $push: { items: item } },
+    ).lean();
+    return Promise.resolve(cart);
+  }
+
+  async findCartByUserId(userId: string): Promise<Cart | null> {
+    const cart = await CartModel.findOne({ user: userId }).lean().lean().exec();
+    return Promise.resolve(cart);
+  }
+
+  async findItemByUserIdAndProductId(
```

Comment:
keep function name simple. `findItem`. take the parameters you need in that function. If you keep function name like this you cant reuse this function when finding is required by itemId